CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+
+7. **Committing your changes**
+
+Now you have made the changes, though they are saved in your system, Git doesn't know what changes you've done. So you have to **commit** your changes. First step is to add the files which you want to add to the staging area, the dot after **add** in the first command tells Git to check for changes in all the files. The second step is about committing your changes. The message part is short description of your commit, like "adds a login form on homepage". Please make sure NOT to have commit messages like "fix issue#34". When we look at the commit history, we should understand what a particular commit is supposed to do based on the commit message.
+
+```
+git add .
+git commit -m "Write message about your commit"
+```
+
+8. **Making sure you have the latest changes from the develop-ember branch**
+
+It may so happen that since the last time you cloned the repo/took a pull from develop, some changes may be merged in the develop-ember branch. So to be on the safer side, we should have those changes as well.
+
+In order to do that, we first checkout to **develop-ember** branch by:
+
+```
+git checkout develop-ember
+```
+
+Once we're in develop-ember, it's time to take a pull:
+
+```
+git pull upstream develop-ember
+```
+
+Now that our **local** develop-ember branch is in sync with **remote** develop-ember branch (of the Real Dev Squad Repository), we should let our branch know about the changes from the develop-ember branch (if any). To do so we first checkout to our branch:
+
+```
+git checkout <branch-name>
+```
+
+Once we're in our branch, we **rebase** our branch on top of the current develop-ember branch (we change the base of our branch, so that it appears as if we have worked from the time the latest changes were merged in the develop-ember branch). To do so:
+
+```
+git rebase develop-ember
+```
+
+You should solve the merge conflicts, if any.
+
+9. **Pushing the code**
+
+Now that we have made our changes plus we have the latest changes made by other contributors, we should push our code from **local** branch to the same branch on our **GitHub fork**. We do so by:
+
+```
+git push origin <branch-name>
+```
+
+For example, if the branch name is `feat/login-form`, we enter `git push origin feat/login-form`
+
+The **origin** refers to your GitHub fork. You can check it by entering `git remote -v`, you should the link to your fork against **origin**.
+
+10. **Making a pull request**
+
+Your GitHub fork now has the changes, but you want those changes to be merged in the Real Dev Squad repository, right? There's a twist, you can't directly merge your code in the Real Dev Squad. Imagine you own a company whose code is open sourced, would you like if anyone could code make changes in the directly without asking you? For the same reason, **Pull Requests** exist. You `request` the repository maintainers/admins to `pull` your code in their repository.
```

Comment:
Sentence can be better formed here. What do you say?
`open sourced, would you like if anyone could make code changes in the repository directly without asking you?`

Reply:
Same here Rohan will read this whole documentation and will ping by EOD.

Reply:
I have rectified and read the documentation once more. Please review it thanks.